### ISSUE TITLE
[BUG] Switching to another person/album shows assets from the previous person/album

### ISSUE THREAD

**Issue by daniele-athome** [immich-app/immich#3823]
### The bug

If you go to the asset grid view for a person, try to open an asset and choose another person in the photo, you'll be taken back to the asset grid view for the new person, but the assets will still be from the previous person.

Refreshing the page works around the issue.

### The OS that Immich Server is running on

Debian

### Version of Immich Server

v1.74.0

### Version of Immich Mobile App

(not applicable)

### Platform with the issue

- [ ] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
(not applicable)
```

### Your .env content

```Shell
(not applicable)
```

### Reproduction steps

```bash
1. go to the Explore page and click on a person
2. from the assets grid view, click an asset to open it
3. open the info panel and click on another person that appears in the photo
4. you'll see the header for the new person, but the assets from the old person
```

**Comment by daniele-athome** [comment]
I just discovered that the same issue (although I don't know if they are related, just guessing) happens with albums:

1. open an album
2. open an asset
3. add asset to shared album
4. create a shared album to add the asset to
5. the view will switch to the new shared album, but the assets inside are the ones from the previous album

**Comment by daniele-athome** [comment]
This seems to be serious upstream:

https://github.com/sveltejs/kit/issues/1497
https://github.com/sveltejs/kit/issues/552
https://github.com/sveltejs/kit/discussions/5007

**Comment by daniele-athome** [comment]
I found a very ugly way to workaround the Svelte bug, but I'm not fluent enough on Svelte, so any Svelte expert willing to lend a hand here would be super nice :-)

https://github.com/daniele-athome/immich/commit/ba7a8af0d182d2f1fbeccc4c5508cf6e91e30866

The key is adding the `#key` block around the AssetGrid otherwise the grid will be stuck on loading mode. But I'm sure there is a better way to do this, there's probably something to be modified in AssetGrid too.

**Comment by daniele-athome** [comment]
Fixed with the workaround by #4089.
